### PR TITLE
[CI] fix spelling mistakes in the codebase by adding codespell

### PR DESCRIPTION
**Pull request by icfaust**
## Description

Root out all the misspellings in the codebase by using ```codespell``` (https://github.com/codespell-project/codespell) a tool currently also in use in oneTBB.  This can be simply added as a pre-commit hook along with our other linters.  This should improve the quality of the codebase and docs at little cost (5 seconds or less). This will improve the overall professional quality we show externally.

This PR includes fixes for all hits which were found (73 files).

No performance benchmarks necessary as it is a lint.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.
- [x] I have provided justification why quality metrics have changed or why changes are not expected.
- [x] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

File: daal4py/sklearn/cluster/dbscan.py [hunk 1]
```diff
@@ -36,10 +36,9 @@ def _daal_dbscan(X, eps=0.5, min_samples=5, sample_weight=None):
     ww = make2d(sample_weight) if sample_weight is not None else None
     XX = make2d(X)
 
-    fpt = getFPType(XX)
     alg = daal4py.dbscan(
         method="defaultDense",
-        fptype=fpt,
+        fptype=getFPType(XX),
```
Comment (david-cortes-intel) [inline daal4py/sklearn/cluster/dbscan.py]: @icfaust Could this be made as an optional check producing some report like the code coverage ones? Otherwise, I don't think it'd be helpful to require changes like this one.

Comment (icfaust) [reply daal4py/sklearn/cluster/dbscan.py]: I could only find this: https://stackoverflow.com/questions/75500530/how-to-avoid-disabling-a-pre-commit-hook so it looks to not be possible.  Let me know why you think not fixing these issues is considered better (given this will reduce review burden).  If something is truly desired there are ways to in a case-by-case basis to disable the check.

Comment (david-cortes-intel) [reply daal4py/sklearn/cluster/dbscan.py]: Could you make this specific case of 'fpt' disable the check as an example here? Could be a good test for the CI logic.

Comment (icfaust) [reply daal4py/sklearn/cluster/dbscan.py]: There will be a file under ```.github/.codespellrc``` which can deselect words for the whole codebase, the whole list is currently : ```als,coo,inout,nd,ons```.  I have changed that specific circumstance to an inline ignore to show the case-by-case basis as was desired.

**Comment by david-cortes-intel** [comment]
@icfaust Since this appears to take a lot longer to run than the usual linting, could it be moved into a separate job?

**Comment by icfaust** [comment]
> @icfaust Since this appears to take a lot longer to run than the usual linting, could it be moved into a separate job?

The last 5 merges to main
#2544 58s (https://dev.azure.com/daal/daal4py/_build/results?buildId=47635&view=results)
#2547 54s (https://dev.azure.com/daal/daal4py/_build/results?buildId=47744&view=results)
#2514 1m 0s (https://dev.azure.com/daal/daal4py/_build/results?buildId=47746&view=results)
#2530 1m 2s (https://dev.azure.com/daal/daal4py/_build/results?buildId=47747&view=results)
#2521 1m 1s (https://dev.azure.com/daal/daal4py/_build/results?buildId=47750&view=results)

The run for this PR: 54s https://dev.azure.com/daal/daal4py/_build/results?buildId=47748&view=results

I don't understand? Could you link me to where you were looking?

**Comment by david-cortes-intel** [comment]
> > @icfaust Since this appears to take a lot longer to run than the usual linting, could it be moved into a separate job?
> 
> The last 5 merges to main #2544 58s (https://dev.azure.com/daal/daal4py/_build/results?buildId=47635&view=results) #2547 54s (https://dev.azure.com/daal/daal4py/_build/results?buildId=47744&view=results) #2514 1m 0s (https://dev.azure.com/daal/daal4py/_build/results?buildId=47746&view=results) #2530 1m 2s (https://dev.azure.com/daal/daal4py/_build/results?buildId=47747&view=results) #2521 1m 1s (https://dev.azure.com/daal/daal4py/_build/results?buildId=47750&view=results)
> 
> The run for this PR: 54s https://dev.azure.com/daal/daal4py/_build/results?buildId=47748&view=results
> 
> I don't understand? Could you link me to where you were looking?

Yes, seems it's fine - I might have been looking at the waiting time which was longer.

**Comment by icfaust** [comment]
gotcha. it seems that a bunch of PRs were merged today to oneDAL and sklearnex all at once so wait times for runners went up quite a lot

File: generator/gen_daal4py.py [hunk 2]
```diff
@@ -972,7 +972,7 @@ def prepare_hlwrapper(self, ns, mode, func, no_dist, no_stream):
                                 itype,
                                 "const",
                                 dflt,
-                                inpt=True,
+                                inp=True,
```
Comment (ethanglaser) [inline generator/gen_daal4py.py:975]: Not so sure about this one

**Review by ethanglaser — COMMENTED** [review]
Great to add this, thanks for doing it! On commented stuff - excellent. On variable names, I am fine to leave or keep as is. On functions and args - we should be careful but looks like theyre used internally here. But I don't think changing inpt->inp is necessary anyway.
